### PR TITLE
More unit tests

### LINKED ISSUE #51: Unit tests
We need unit tests.

### PR DESCRIPTION
**Pull request by Innectic**
Added unit tests for ban packet, and a few commands. It's not worth adding tests for the other commands, since we're switching to doc tests in 0.4.1. Fixes #51

### CODE REVIEW

File: tests/handlers/test_respond.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+response_handler = ResponseHandler()
+
+
+@pytest.mark.asyncio
+async def test_user_update():
+    """Test the user update event."""
+
+    assert (await response_handler.on_username_update(Packet(username="TestUser"))
+           ) is None
```
Comment (2Cubed) [inline tests/handlers/test_respond.py]: The fact that this returns `None` doesn't tell us too much in this case. We should also have `assert response_handler.username == "TestUser"` afterwards. :slightly_smiling_face:

File: tests/packets/test_message.py [hunk 2]
```diff
@@ -9,12 +9,28 @@ def _split(text, *args, **kwargs):
...
 def test_split():
+    """Test splitting message packets."""
 
     assert _split("0 1 2 3") == ['0', '1', '2', '3']
-
     assert _split("0 1 2 3", "2") == ['0 1 ', ' 3']
-
     assert _split("0 1 2 3", maximum=2) == ['0', '1', '2 3']
-
     assert _split("0 1 2 3 ") == ['0', '1', '2', '3']
```
Comment (2Cubed) [inline tests/packets/test_message.py]: Can you add a test for `" 0 1 2 3"` and `" 0 1 2 3 "`? My bad.

File: tests/packets/test_message.py [hunk 3]
```diff
@@ -9,12 +9,28 @@ def _split(text, *args, **kwargs):
...
+
+
+def test_join():
+    """Test joining message packets."""
+
+    packet1 = MessagePacket("I like ")
+    packet2 = MessagePacket("kittens!")
+
+    assert MessagePacket.join(packet1, packet2).text == "I like kittens!"
```
Comment (2Cubed) [inline tests/packets/test_message.py]: Nice! We should have a few more tests, though, *especially* looking for edge cases. We can assume that the feature works as intended for simple cases (here, joining two text packets), but bugs often come from weird things, just as we saw with the `MessagePacket.split` having a space at the end.

Do we get what we want with:
- no packets
- one/more empty packet(s) joined
- packets containing more than text (emoji? tag? url?)
- etc.

File: cactusbot/commands/magic/alias.py [hunk 4]
```diff
@@ -14,6 +14,10 @@ class Alias(Command):
                   raw: "packet"):
         """Add a new command alias."""
 
+        print(alias)
+        print(command)
+        print(*_)
+        print(raw)
```
Comment (Innectic) [inline cactusbot/commands/magic/alias.py]: I'm really bad at remembering to remove my test code.

File: tests/packets/test_message.py (hunk 3 above)
Comment (Innectic) [inline tests/packets/test_message.py]: I didn't add any of the edge cases because this is all going away in the next version.

Comment (2Cubed) [inline tests/packets/test_message.py]: Wait... what? o.O

Comment (Innectic) [inline tests/packets/test_message.py]: #169

Comment (2Cubed) [inline tests/packets/test_message.py]: Oh, gosh, we're not moving *all* unit tests to doctests... just those for magic commands, specifically. These will be staying.

```python
>>> Cactus("github", "sepal").text
"https://github.com/CactusDev/Sepal"
```
